tests/integration/test_s3.py

```diff
@@ -282,6 +284,14 @@ def test_s3_batch_delete(self):
         versions = list(bucket.object_versions.all())
         self.assertEqual(len(versions), 0)
 
+    def test_unsigned_access(self):
```

Comment:
What is your thoughts on just adding a test to make sure `s3.meta.client.meta.config.signature_version is botocore.UNSIGNED`? That seems to be the bug for this case.

I think if we are going to be adding an integration test for this I would prefer that we are asserting somehow `UNSIGNED` is being used, as while it does ensure we do not have a general runtime error, it is sort of vague in terms of what the desired outcome is as a ClientError can be raised for various reasons. Maybe make a publicly readable object and try to download using `UNSIGNED`? Or check the error to make sure it is something about the request not being authorized/unsigned?